### PR TITLE
Assessing OSM's sufficiency and calculating needed features

### PR DESCRIPTION
**Pull request by tm-abby-moreno**
## What does this PR do?

1. Goal of this PR was to assess how complete and sufficient is OSM for the project while also establishing calculations/functions in creating features. Features that were calculated in this PR were: total POI counts (of each type), nearest distances, and also distance to the coast.
2. Establish file organization for the repository! I tried to separate the utils to `py` scripts instead of placing them in the notebook and I also arranged the notebooks and data folder. Open to more suggestions on organization and file conventions. 

## Where should the reviewers start?

- `00-compute-coastal-buffer.ipynb` - Notebook used for generating the coastal buffer. It computes for a one-sided buffer based on the custom function from the [CI project](https://github.com/thinkingmachines/aquafarm-mapping-site-scoring/blob/main/notebooks/phl_pipeline/05-table_tfm/generate_coastal_buffer.ipynb). Did not attempt to use something else for generating the coastal buffer since it might take time and would prefer not to overengineer for this part.
- `01-OSM-assessment.ipynb` - **Important notebook to check!!!** Main notebook that was worked on. Contains creation of features, plots, and other exploratory data analyses. Open to more suggestions on EDA (if needed)! It's a pretty big dataset to cover but I tried my best to cover as much info we needed to move forward. 
- `src/vector_utils.py` - Contains utils that were used in calculating the said features above. For the calculation of the counts and nearest distances for polygons, these are just based on the [OSM povertymapping util](https://github.com/thinkingmachines/unicef-ai4d-poverty-mapping/blob/main/povertymapping/osm.py). Separated them from the notebook to be more organized. Also open to more suggestions on organizing utils! 
- Check overall file and notebook organization! Open to feedback on this since my notebooks tend to get a little messy at times. Would like to fix this as early as now since this will be a long project :DD

### CODE REVIEW

**Comment by tm-abby-moreno** [comment]
Addressed almost everything!! I updated some cells regarding the formats and the column names, but I have not ran the whole notebook yet and exported the files since you'll take a second pass at this with reviewNB @tm-jace-peralta ! I left 2 things unaddressed because of a few thoughts:

- I added the `osm_year` but not the `datetime_created` because I don't think it would be relevant for people to know the specific datetime we ran the processing functions. Can I ask what the `datetime_created` would be used for?
- For saving copies of both `csv` and `geojson`, I'm thinking of just saving them all to `csv`'s without the geometry column since it will be redundant and might take up too much space. Was thinking of just leaving the `psgc_codes` + relevant features for each dataset when saving them and then join them to the admin boundaries table at the end to create the `geojson`.

**Comment by tm-jace-peralta** [comment]
Hi Abby,

1. On `datetime_created` - dataset sources may issue errata/updates to existing files sometime in the future, it is important for users to know when the data was created to be able to trace if our linked dataset is affected by these possible corrections. 
2. Point taken on storage space, I agree with you, lets save to ` .csv` only and just keep code to save to` .geojson `

File: src/osm_poi_categories.json [hunk 1]
```diff
@@ -0,0 +1,12 @@
+{
+    "health": ["clinic", "dentist", "doctors", "hospital","optician", "pharmacy"],
+    "crit_lifelines": ["atm","bank", "college", "community_centre", "comms_tower", "convenience",
```
Comment (tm-jace-peralta) [inline src/osm_poi_categories.json:3]: seems like 'crit_lifelines' is not the optimal label for this subset of poi categories. 

- move 'comms_tower', 'telephone' to `telco`  label
- remove tower, already in `power` category
- add new label `commercial`: ['atm', 'bank', 'convenience ;store', 'market_place', 'supermarket'] 
- add new label `public:` [remaining categories in crit_lifelines]

File: src/vector_utils.py [hunk 2]
```diff
@@ -0,0 +1,245 @@
...
+import geopandas as gpd
+from geopandas import GeoDataFrame
+from geowrangler import distance_zonal_stats as dzs
+from geowrangler import vector_zonal_stats as vzs
+from loguru import logger
+from shapely.geometry import GeometryCollection, MultiPolygon, Polygon
+from shapely.geometry.polygon import orient
+from shapely.validation import explain_validity, make_valid
+
+# Set directories
```
Comment (tm-jace-peralta) [inline src/vector_utils.py:13]: remove unnecessary directory declarations here

**Review by tm-jace-peralta — APPROVED** [review]

**Pull request merged**